src/inference/src/dev/core_impl.cpp

```diff
@@ -877,7 +881,11 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
         CacheContent cacheContent{cacheManager, parsed._core_config.get_enable_mmap()};
         cacheContent.blobId = ov::ModelCache::compute_hash(model, create_compile_config(plugin, parsed._config));
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
-        cacheContent.model = model;
+        if (auto mode_it = parsed._config.find(ov::cache_mode.name()); mode_it != parsed._config.end()) {
+            if (mode_it->second.as<ov::CacheMode>() == ov::CacheMode::OPTIMIZE_SIZE) {
+                cacheContent.model = model;
```

Comment:
I don't understand why do we need it?
Why GPU plugin holds ov::Model in such case?